src/Elyon.Fastly.Api.DomainServices/InfoSessionFollowUpService.cs

```diff
@@ -49,18 +53,128 @@ public async Task SendInfoSessionFollowUpEmailAsync(InfoSessionFollowUpSpecDto s
             var organization = await _organizationsRepository
                 .GetByIdAsync(specDto.OrganizationId).ConfigureAwait(false);
 
-            if (!ValidateOrganizationForNewFollowUp(organization))
-                return;
+            if (organization.OrganizationType.Id == _smeOrganizationId)
+            {
+                if (!specDto.SMSStartDate.HasValue || specDto.SMSStartDate == default(DateTime))
+                {
+                    ValidationDictionary.AddModelError("SMS Start Date",
+                        "SMS Start Date is required.");
+                    return;
+                }
+                if (!ValidateOrganizationForOnboarding(organization))
+                    return;
 
-            string generatedToken = await AddFollowUpAsync(organization).ConfigureAwait(false);
+                foreach (var receiver in specDto.Receivers)
+                {
+                    var parameters = new Dictionary<string, string> 
+                    {
+                        { "SMSdate", ((DateTime)specDto.SMSStartDate).ToString("d", CultureInfo.CreateSpecificCulture("de-CH")) },
+                        { "OnboardingDate", ((DateTime)organization.OnboardingTimestamp).ToString("d", CultureInfo.CreateSpecificCulture("de-CH")) },
+                        { "areaPharmacy", organization.Area },
+                        { "CompanyShortcut", organization.OrganizationShortcutName },
+                        { "firstTestingDate", ((DateTime)organization.FirstTestTimestamp).ToString("d", CultureInfo.CreateSpecificCulture("de-CH")) },
+                        { "TestingDay", ((DateTime)organization.FirstTestTimestamp).ToString("dddd", CultureInfo.CreateSpecificCulture("de-CH")) }
```

Comment:
are you sure you need to use the organization.FirstTestTimestamp for the TestingDay?

Reply:
Yes, I confirmed one more time with Viktor.